client/app/house/components/MembersManagementModal.tsx

```diff
@@ -0,0 +1,149 @@
+import { CopyToClipboard } from 'react-copy-to-clipboard'
+import { FiPaperclip, FiX } from 'react-icons/fi'
+import { MdAccountCircle } from 'react-icons/md'
+import { Button, Checkbox } from '@/_components/ui'
+import {
+  Dialog,
+  DialogContent,
+  DialogHeader,
+  DialogTitle,
+} from '@/_components/ui/dialog/Dialog'
+import { colorMap } from '@/_consts'
+import { useToast } from '@/_hooks'
+import { HouseType } from '@/_types'
+import { useUpdateAdmin } from '../hooks'
+
+type MembersManagementModalProps = {
+  house: HouseType
+  isOpenModal: boolean
+  setIsOpenModal: (isOpenModal: boolean) => void
+  isOpenAlert: boolean
+  setIsOpenAlert: (isOpenAlert: boolean) => void
+  removedMemberId: number | null
+  setRemovedMemberId: (removedMemberId: number | null) => void
+}
+
+const MembersManagementModal = ({
+  house,
+  isOpenModal,
+  setIsOpenModal,
+  setIsOpenAlert,
+  setRemovedMemberId,
+}: MembersManagementModalProps) => {
+  const { toast } = useToast()
+  const { onUpdate, isPending: isUpdating } = useUpdateAdmin()
+  const houseId = house.houseId
+  const [firstMember, ...restMembers] = house.houseMembers
+  const sortedRestMembers = restMembers.sort((a, b) => a.id - b.id) // Sorted by userId to fix the order
+  const houseMembers = [firstMember, ...sortedRestMembers]
+  const userId = houseMembers[0].id
+  const invitedLink =
+    process.env.NEXT_PUBLIC_CLIENT_URL + `/register?invitedHouseId=${houseId}`
+
+  const handleUpdateAdmin = (userId: number, isAdmin: boolean) => {
+    const data = {
+      userId,
+      houseId,
+      isAdmin,
+    }
+
+    onUpdate(data)
+  }
+
+  const handleOpenAlert = (userId: number) => {
+    setRemovedMemberId(userId)
+    setIsOpenModal(false)
+    setIsOpenAlert(true)
+  }
+
+  return (
+    <Dialog open={isOpenModal} onOpenChange={() => setIsOpenModal(false)}>
+      <DialogContent
+        // To prevent closing modal when clicking toast
+        onPointerDownOutside={(e) => {
+          if (
+            e.target instanceof Element &&
+            e.target.closest('[data-radix-toast-announce-exclude]')
+          ) {
+            e.preventDefault()
+          }
+        }}
+      >
+        <DialogHeader className='pb-2'>
+          <DialogTitle className='text-xl'>Members management</DialogTitle>
+        </DialogHeader>
+        <div className='hide-scrollbar flex max-h-96 flex-col overflow-scroll'>
```

Comment:
If the name of the user is long, the icon shrinks, but we don't want that happen. And maybe the gap between the checkbox and `admin` is too much for SP? I feel the text overall is too big such as the `Role`, `Remove`, the other labels for SP!

<img width="425" alt="Screenshot 2024-04-02 at 4 22 32 PM" src="https://github.com/HAK2024/fairy-share/assets/66394413/a4c173a5-7420-4f37-948d-e86e28d20337">

Reply:
Fixed!
https://github.com/HAK2024/fairy-share/pull/93/commits/a3993a503b2e4980fccae76f7e62f0c8f3b885ee

Reply:
The vertical line does not align, could you fixt it? I also thought you wouldn't have to hide the name if it was too long. You can show like the second screen shot. (There is too much space though) 
<img width="478" alt="Screenshot 2024-04-03 at 4 11 06 PM" src="https://github.com/HAK2024/fairy-share/assets/66394413/d22351f7-c42a-42df-a3a4-341c9cab8254">


<img width="612" alt="Screenshot 2024-04-03 at 4 11 51 PM" src="https://github.com/HAK2024/fairy-share/assets/66394413/a6bb73eb-441c-4183-8fcb-13fea3a65e9e">

Reply:
Fixed it!
Unless you made screen size under 370px, it wont break the design!
https://github.com/HAK2024/fairy-share/pull/93/commits/5f776da836a151fe042fb8599c353dd3c567f150
https://github.com/HAK2024/fairy-share/pull/93/commits/4430dac8d7ed3c486302c1ddabaeb9d138179c48